tests/e2e/utils/kubeflow_installation.py

```diff
@@ -0,0 +1,200 @@
+from pickle import FALSE
+from e2e.utils.utils import load_yaml_file, print_banner
+import argparse
+from e2e.utils.utils import (
+    kubectl_wait_pods,
+    apply_kustomize,
+    install_helm,
+)
+import subprocess
+import os
+
+INSTALLATION_PATH_FILE_VANILLA = "./resources/installation_config/vanilla.yaml"
+INSTALLATION_PATH_FILE_COGNITO = "./resources/installation_config/cognito.yaml"
+
+Install_Sequence = [
+    "cert-manager",
+    "kubeflow-roles",
+    "kubeflow-issuer",
+    "istio-1-14",
+    "kubeflow-namespace",
+    "dex",
+    "cluster-local-gateway",
+    "knative-serving",
+    "knative-eventing",
+    "oidc-authservice",
+    "kubeflow-istio-resources",
+    "kserve",
+    "models-web-app",
+    "central-dashboard",
+    "aws-secrets-manager",
+    "kubeflow-pipelines",
+    "admission-webhook",
+    "jupyter-web-app",
+    "notebook-controller",
+    "volumes-web-app",
+    "training-operator",
+    "katib",
+    "tensorboards-web-app",
+    "tensorboard-controller",
+    "profiles-and-kfam",
+    "user-namespace",
+    #"ack-sagemaker-controller",
+    "ingress",
+    "aws-load-balancer-controller",
+    "aws-authservice",
+]
+
+
+def install_kubeflow(
+    installation_option, aws_telemetry_option, deployment_option, cluster_name
+):
+    INSTALLATION_OPTION = installation_option
+    AWS_TELEMETRY_OPTION = aws_telemetry_option
+    DEPLOYMENT_OPTION = deployment_option
+
+    if DEPLOYMENT_OPTION == "vanilla":
+        path_dic = load_yaml_file(INSTALLATION_PATH_FILE_VANILLA)
+    elif DEPLOYMENT_OPTION == "cognito":
+        path_dic = load_yaml_file(INSTALLATION_PATH_FILE_COGNITO)
+
+    print_banner(
+        f"You are installing kubeflow {DEPLOYMENT_OPTION} deployment with {INSTALLATION_OPTION}"
+    )
+
+    for component in Install_Sequence:
+        build_component(
+            INSTALLATION_OPTION,
+            DEPLOYMENT_OPTION,
+            component,
+            path_dic,
+            cluster_name,
+        )
+
+    if AWS_TELEMETRY_OPTION == "enable":
+        build_component(
+            INSTALLATION_OPTION,
+            DEPLOYMENT_OPTION,
+            "aws-telemetry",
+            path_dic,
+            cluster_name,
+        )
+
+
+def build_component(
+    INSTALLATION_OPTION,
+    DEPLOYMENT_OPTION,
+    component_name,
+    path_dic,
+    cluster_name,
+    crd_meet=True,
+    namespace=None,
+):
+    print(f"==========Installing {component_name}==========")
+    if component_name not in path_dic:
+        print(
+            f"component {component_name} is not applicable for deployment option: {DEPLOYMENT_OPTION}"
+        )
+        return
+    else:
+        installation_path = path_dic[component_name]["installation_options"][
+            INSTALLATION_OPTION
+        ]
+
+        if INSTALLATION_OPTION == "helm":
+            # cert-manager official chart command call
+            if component_name == "cert-manager":
+                build_retcode = build_certmanager()
```

Comment:
nit: build -> install_component_helm

Reply:
what do you mean by this? you mean make a function for "cert-manager" and "aws-load-balancer-controller"?

Reply:
I mean the name of function build_certmanager -> install_certmanager_helm

Reply:
addressed